lib/iris/_shapefiles.py

```diff
@@ -0,0 +1,223 @@
+                            eg: 0.1 means that at least 10% of the shape overlaps the cell
+                            to be unmasked
+
+    Returns:
+        A numpy array of the shape of the x & y coordinates of the cube, with points to mask equal to True
+    """
+
+    from iris.cube import Cube, CubeList
+
+    try:
+        msg = TypeError("Geometry is not a valid Shapely object")
+        if geometry.is_valid is False:
+            raise msg
+    except Exception:
+        raise msg
+    if not isinstance(cube, Cube):
+        if isinstance(cube, CubeList):
+            msg = "Received CubeList object rather than Cube - \
+            to mask a CubeList iterate over each Cube"
+            raise TypeError(msg)
+        else:
+            msg = "Received non-Cube object where a Cube is expected"
+            raise TypeError(msg)
+    if not minimum_weight or minimum_weight == 0.0:
+        weights = False
+    elif isinstance(
+        geometry,
+        (
+            sgeom.Point,
+            sgeom.LineString,
+            sgeom.LinearRing,
+            sgeom.MultiPoint,
+            sgeom.MultiLineString,
+        ),
+    ):
+        weights = False
+        warnings.warn(
+            """Shape is of invalid type for minimum weight masking,
+            must use a Polygon rather than Line shape.\n
+              Masking based off intersection instead. """,
+            IrisDefaultingWarning,
+        )
+    else:
+        weights = True
+
+    # prepare shape
+    trans_geo = _transform_coord_system(geometry, cube, shape_coord_system)
+    prepped_trans_geo = prepped.prep(trans_geo)
+
+    # prepare 2D cube
+    y_name, x_name = _cube_primary_xy_coord_names(cube)
+    cube_2d = cube.slices([y_name, x_name]).next()
+    _cube_xy_guessbounds(cube_2d)
+    xmod = cube.coord(x_name).units.modulus
+    ymod = cube.coord(y_name).units.modulus
+    mask_template = np.zeros(cube_2d.shape, dtype=bool)
+
+    # perform the masking
+    for count, idx in enumerate(np.ndindex(cube_2d.shape)):
+        # get the bounds of the grid cell
+        xi = idx[cube_2d.coord_dims(x_name)[0]]
+        yi = idx[cube_2d.coord_dims(y_name)[0]]
+        x0, x1 = cube_2d.coord(x_name).bounds[xi]
+        y0, y1 = cube_2d.coord(y_name).bounds[yi]
+
+        if xmod:
+            x0, x1 = _rebase_values_to_modulus((x0, x1), xmod)
+        if ymod:
+            y0, y1 = _rebase_values_to_modulus((y0, y1), ymod)
+
+        # create a new polygon of the grid cell and check intersection
+        cell_box = sgeom.box(x0, y0, x1, y1)
+        intersect_bool = prepped_trans_geo.intersects(cell_box)
+        # mask all points without a intersection
+        if intersect_bool is False:
+            mask_template[idx] = True
+        # if weights method used, mask intersections below required weight
+        if intersect_bool is True and weights is True:
+            intersect_area = trans_geo.intersection(cell_box).area
```

Comment:
```suggestion
            intersect_area = prepped_trans_geo.intersection(cell_box).area
```

?

Reply:
a prepared geometry doesn't have the "intersection" method, but it does have the "intersects" method. (one gives a boolean, the other gives the whole shape of the intersection, hence the slowness).